src/Estimators/src/RecursiveLeastSquare.cpp

```diff
@@ -0,0 +1,72 @@
+/**
+ * @file RecursiveLeastSquare.cpp
+ * @authors Giulio Romualdi
+ * @copyright 2020 Istituto Italiano di Tecnologia (IIT). This software may be modified and
+ * distributed under the terms of the GNU Lesser General Public License v2.1 or any later version.
+ */
+
+#include <iDynTree/Core/EigenHelpers.h>
+
+#include <BipedalLocomotionControllers/Estimators/RecursiveLeastSquare.h>
+
+using namespace BipedalLocomotionControllers::Estimators;
+
+void RecursiveLeastSquare::setRegressorFunction(std::function<iDynTree::MatrixDynSize(void)> regressor)
+{
+    m_regressor = regressor;
+}
+
+bool RecursiveLeastSquare::advance()
+{
+    using iDynTree::toEigen;
+
+    if (m_regressor == nullptr)
+    {
+        std::cerr << "[RecursiveLeastSquare::advance] Please call the setRegressorFunction() "
+                     "before calling advance"
+                  << std::endl;
+        return false;
+    }
+
+    if (m_estimatorState != State::Initialized && m_estimatorState != State::Running)
+    {
+        std::cerr << "[RecursiveLeastSquare::advance] Please initialize the estimator before "
+                     "calling advance."
+                  << std::endl;
+        return false;
+    }
+
+    if (m_estimatorState == State::Initialized)
+        m_estimatorState = State::Running;
+
+    iDynTree::MatrixDynSize regressor = m_regressor();
+    toEigen(m_kalmanGain) = toEigen(m_stateCovarianceMatrix) * toEigen(regressor).transpose()
+                            * (m_lambda * toEigen(m_measurementCovarianceMatrix)
+                               + (toEigen(regressor) * toEigen(m_stateCovarianceMatrix)
+                                  * toEigen(regressor).transpose())).inverse();
```

Comment:
That was pretty verbose.
But it reads, $K = P H^T (\lambda R + H P H^T)^{-1}$ which seems to be coherent.

Reply:
Could you please tell me how $\lambda$ enters the Kalman gain equations?

Reply:
Unfortunately, the Kalman filter interpretation is valid only if `lambda = 1`

Check the following explanation taken from **Lennart Ljung - System Identification: Theory for the User**

![image](https://user-images.githubusercontent.com/16744101/76450241-bd15f600-63cd-11ea-9bc3-0f32b9706129.png)

Actually I don't know if `lambda` different from 1 is really useful.

Reply:
Seems to be working fine from the plots, though ;D

Reply:
In the plots lambda is equal to 1 :smile: